### PR TITLE
Implement support for more `time` types

### LINKED ISSUE #1281: Support more types from the `time` crate
From what I noticed in [`time.rs`](https://github.com/rusqlite/rusqlite/blob/master/src/types/time.rs) only the `time::OffsetDateTime` type is supported by default. 

I'd like to implement support for the other following types:
* `time::Date`
* `time::Time`
* `time::Instant`
* `time::Duration`
* `time::PrimitiveDateTime`
* `time::UtcOffset`
* `time::Weekday`
* `time::Month`

Is this something that you'd accept as a PR?

### PR DESCRIPTION
**Pull request by czocher**
This PR implements support for the following types:
* `time::Time`
* `time::Date`
* `time::PrimitiveDateTime`

Closes #1281

### CODE REVIEW

**Comment by Czocher** [comment]
@gwenn please check if it's up to your expectations :).

File: src/types/time.rs [hunk 1]
```diff
@@ -84,10 +196,10 @@ mod test {
         };
 
         ts_vec.push(make_datetime(10_000, 0)); //January 1, 1970 2:46:40 AM
-        ts_vec.push(make_datetime(10_000, 1000)); //January 1, 1970 2:46:40 AM (and one microsecond)
+                                               // ts_vec.push(make_datetime(10_000, 1000)); //January 1, 1970 2:46:40 AM (and one microsecond)
```
Comment (gwenn) [inline src/types/time.rs:199]: Why ?

File: src/types/time.rs [hunk 2]
```diff
@@ -84,10 +196,10 @@ mod test {
...
         ts_vec.push(make_datetime(1_500_391_124, 1_000_000)); //July 18, 2017
         ts_vec.push(make_datetime(2_000_000_000, 2_000_000)); //May 18, 2033
-        ts_vec.push(make_datetime(3_000_000_000, 999_999_999)); //January 24, 2065
+        ts_vec.push(make_datetime(3_000_000_000, 999)); //January 24, 2065
```
Comment (gwenn) [inline src/types/time.rs:202]: Why ?

Comment (Czocher) [reply src/types/time.rs:202]: Sorry leftover from when I tried to rewrite the `OffsetDateTime` implementation. Removed this unnecessary change.

File: src/types/time.rs (hunk 1 above)
Comment (Czocher) [reply src/types/time.rs:199]: Sorry leftover from when I tried to rewrite the `OffsetDateTime` implementation. Uncommented the code.

File: src/types/time.rs (hunk 2 above)
Comment (gwenn) [reply src/types/time.rs:202]: Someone else fixes it here: #1374

File: src/types/time.rs (hunk 1 above)
Comment (gwenn) [reply src/types/time.rs:199]: Someone else fixes it here: #1374

**Comment by gwenn** [comment]
This abandoned PR has been fixed here: #1374

**Pull request closed without merging**